### ISSUE TITLE
Weird behavior with WriteHeader

### ISSUE THREAD

**Issue by julian-goldsmith** [JoshClose/CsvHelper#798]
I just got my project to compile with CsvHelper 4.0, and I think I've found a problem with WriteHeader.

```
class CsvHelperTestData
{
	public string TestStringData { get; set; }
	public int TestIntData { get; set; }
}

public void Main()
{
	CsvHelperTestData tdata = new CsvHelperTestData
	{
		TestStringData = "Test string for CsvHelperTest",
		TestIntData = 1234
	};
	
	using (MemoryStream ms = new MemoryStream())
	{
		using (TextWriter tw = new StreamWriter(ms, Encoding.UTF8, 1024, true))
		using (CsvWriter cw = new CsvWriter(tw))
		{
			cw.WriteHeader<CsvHelperTestData>();
			//cw.Configuration.HasHeaderRecord = true;
			cw.WriteRecords(new [] {tdata});
			tw.Close();
		}
		
		ms.Position = 0;
		
		using (TextReader tr = new StreamReader(ms))
		{
			tr.ReadToEnd().Dump();
		}
	}
}
```

In CsvHelper 2, the above code would produce two lines: one for the header, and one for the data.  In CsvHelper 4, it adds the data to the header row.  If you get rid of the WriteHeader call, and use HasHeaderRecord, the data is written as expected.

**Comment by JoshClose** [comment]
This is correct behavior, and I'll explain what's going on. I may need to add some more info the the docs. http://joshclose.github.io/CsvHelper/writing

When you use `WriteRecords`, it will write the header for you if one hasn't already been written. Since you're writing the header by hand you'll need to call `NextRecord` to end writing to that row.

Out of curiosity, why would you need to write the header separately when using `WriteRecords`?

**Comment by julian-goldsmith** [comment]
Ah, I didn't know that.  That makes sense, though.  Thanks for the help.

The original code was taken out of a unit test for a `TextWriter` implementation, and used `WriteRecord` to write a single record.  When `WriteRecord` is used, it doesn't write a header record, so I was writing it manually.  Older versions of CsvHelper wrote the header as its own row, so my unit test failed when I updated.  The code I posted here was taken from a LINQPad script, where I had changed the `WriteRecord` call to `WriteRecords`.
